internal/generate/types.go

```diff
@@ -511,8 +513,13 @@ func createTypeObject(schema *openapi3.Schema, name, typeName, description strin
 			}
 		}
 
-		// If a type is nullable we'll want a pointer
-		if sliceContains(nullable(), typeName) {
+		// Omicron includes fields that are both required and nullable:
+		// they can be set to a null value, but they must not be
+		// omitted. The sdk should present these fields to the user as
+		// optional, and serialize them to `null` if not provided.
+		isRequired := slices.Contains(required, k)
+		isRequiredNullable := v.Value.Nullable && isRequired
+		if (slices.Contains(nullable(), typeName) || isRequiredNullable) && !strings.HasPrefix(typeName, "*") {
```

Comment:
What's the motivation for `!strings.HasPrefix(typeName, "*")`? Is it to exclude pointer types? If future me were reading this I wouldn't know what the significance of this is.

Reply:
Yeah, sometimes by this point we've already decided to use a pointer for the type, so its `typeName` already begins with "*". This is a bit of a hack, but I don't want to take on a bigger refactor just yet. For now, I'm just adding a comment here explaining why we're checking the prefix.

Reply:
Okay. That's fine with me.